korman/nodes/node_avatar.py

```diff
@@ -71,3 +72,393 @@ def export(self, exporter, bo, so):
+
+    @property
+    def stage_id(self):
+        idx = None
+        stage_socket = self.find_output_socket("stage")
+        if stage_socket.is_linked:
+            msbmod = stage_socket.links[0].to_node
+            idx = next((idx for idx, socket in enumerate(msbmod.find_input_sockets("stage_refs")) if socket.is_linked and socket.links[0].from_node == self))
+        return idx
+
+
+class PlasmaBehaviorSocket(PlasmaNodeSocketBase, bpy.types.NodeSocket):
+    bl_color = (0.348, 0.186, 0.349, 1.0)
+
+
+class PlasmaMultiStageBehaviorNode(PlasmaNodeBase, bpy.types.Node):
+    bl_category = "AVATAR"
+    bl_idname = "PlasmaMultiStageBehaviorNode"
+    bl_label = "Multistage Behavior"
+    bl_width_default = 200
+
+    pl_attrib = ("ptAttribBehavior")
+
+    freeze_phys = BoolProperty(name="Freeze Physical",
+                              description="Freeze physical at end",
+                              default=False)
+    reverse_control = BoolProperty(name="Reverse Controls",
+                              description="Reverse forward/back controls at end",
+                              default=False)
+
+    input_sockets = OrderedDict([
+        ("seek_target", {
+            "text": "Seek Target",
+            "type": "PlasmaSeekTargetSocketIn",
+            "valid_link_sockets": {"PlasmaSeekTargetSocketOut"},
+        }),
+        ("stage_refs", {
+            "text": "Stage",
+            "type": "PlasmaAnimStageRefSocket",
+            "valid_link_nodes": "PlasmaAnimStageNode",
+            "valid_link_sockets": "PlasmaAnimStageRefSocket",
+            "link_limit": 1,
+            "spawn_empty": True,
+        }),
+    ])
+
+    output_sockets = OrderedDict([
+        ("hosts", {
+            "text": "Host Script",
+            "type": "PlasmaBehaviorSocket",
+            "valid_link_nodes": {"PlasmaPythonFileNode"},
+            "spawn_empty": True,
+        })
+    ])
+
+    def draw_buttons(self, context, layout):
+        layout.prop(self, "freeze_phys")
+        layout.prop(self, "reverse_control")
+
+    def get_key(self, exporter, so):
+        return self._find_create_key(plMultistageBehMod, exporter, so=so)
+
+    def export(self, exporter, bo, so):
+        seek_socket = self.find_input_socket("seek_target")
+
+        if seek_socket.is_linked:
+            smartSeek = True
+            seek_target = seek_socket.links[0].from_node.target
+            if seek_target is None:
+                raise ExportError("'{}': MultiStage Behavior's seek point object is invalid".format(self.get_key(exporter, so).name))
+        elif seek_socket.auto_target:
+            smartSeek = True
+            seek_target = bo
+        else:
+            smartSeek = False
+            seek_target = bo
+
+        if smartSeek:
+            spm = self._find_create_object(plSeekPointMod, exporter, bl=seek_target)
+            spm.name = seek_target.name
+
+        msbmod = self._find_create_object(plMultistageBehMod, exporter, bl=seek_target)
```

Comment:
I somehow missed that this still doesn't use `get_key` like it should...

Reply:
`_find_create_object` doesn't take a key, does it?